design/0001-frr.md

```diff
@@ -0,0 +1,391 @@
+                                     │                    │
+                              ┌──────┴────────┐  ┌────────┴──────┐
+                              │     main::    │  │    main::     │
+                              │ L2_controller │  │ bgp_controller│
+                              │   <<class>>   │  │   <<class>>   │
+                              │               │  │               │
+                              └───────────────┘  └───────┬───────┘
+                                                         │
+                                                         │  <<use>>
+                                                 ┌───────▼───────┐
+                                                 │               │
+                                                 │    session    │
+                                                 │ <<interface>> │
+                                                 │               │
+                                                 └───────▲───────┘
+                                                         │ <<implement>>
+                                             ┌───────────┴─────────────┐
+                                             │                         │
+                                     ┌───────┴───────┐         ┌───────┴───────┐
+                                     │               │         │               │
+                                     │  bgp_metallb  │         │    bgp_frr    │
+                                     │  <<package>>  │         │  <<package>>  │
+                                     │               │         │               │
+                                     └───────────────┘         └───────┬───────┘
+                                                                       │
+                                                                       │
+                                                               ┌───────┴───────┐
+                                                               │               │
+                                                               │      frr      │
+                                                               │ <<container>> │
+                                                               │               │
+                                                               └───────────────┘
+
+```
+
+#### FRR deployment
+
+FRR will be deployed as a container. The current proposal is to deploy it
+as a DaemonSet in the MetalLB namespace. The speaker application will be
+configured at startup to use the FRR BGP daemon and will provide details for
+connection to the daemon (e.g. location of a socket).
+
+The container will expose a control interface that will initially be used to
+configure FRR via the `vtysh` application and eventually via its gRPC golang
+bindings. This will be used to implement bgp.New() and bgp.Close() for session
+creation and bgp.Set() for prefix advertisement.
+
```

Comment:
Do you have any idea to gather bgpd logging? At least, for troubleshooting, we need to gather log from bgpd somehow and provide user to look the logs. One candidate could be stdout and user can see through `kubectl logs` command...

Reply:
Yeah, I think this would be the natural way to do this. We can configure FRR using something like `log stdout LEVEL`.

Reply:
Shall we note it in this doc? or it is just taken care of implementation?